src/queue.js

```diff
@@ -0,0 +1,34 @@
+'use strict'
+
+export default class Queue {
+  constructor() {
+    this.head = 0
+    this.sequence = []
+  }
+
+  enqueue( data ) {
+    this.sequence[ this.head ] = data
+    this.head++
+  }
+
+  dequeue() {
+    return this.head > 0 ? this.sequence.splice(0, 1)[0] : null
```

Comment:
Great use of builtins on array to implement this.  Consider rewriting to use `Node` objects, in order to better understand the `Queue` operations.